### PR TITLE
Remove all references to the old semantic analyzer.

### PR DESCRIPTION
**Pull request by msullivan**
Follow-up diffs will delete the code and move the new analyzer

### CODE REVIEW

File: mypy/typeops.py [hunk 1]
```diff
@@ -8,11 +8,12 @@
 # TODO: Move more type operations here
 
 from mypy.types import TupleType, Instance
-from mypy.join import join_type_list
 
 
 def tuple_fallback(typ: TupleType) -> Instance:
     """Return fallback type for a tuple."""
+    from mypy.join import join_type_list
```
Comment (JukkaL) [inline mypy/typeops.py]: What's this change?

File: test-data/unit/check-incremental.test [hunk 2]
```diff
@@ -3338,7 +3340,7 @@ class C:
     d: int = attr.ib(converter=parse)
 [builtins fixtures/attr.pyi]
 [out1]
-main:6: note: Revealed type is 'def (a: Union[builtins.float, builtins.str], b: Union[builtins.str, builtins.bytes, builtins.int], c: builtins.str, d: Union[builtins.int, builtins.str]) -> a.C'
+5main:6: note: Revealed type is 'def (a: Union[builtins.float, builtins.str], b: Union[builtins.str, builtins.bytes, builtins.int], c: builtins.str, d: Union[builtins.int, builtins.str]) -> a.C'
```
Comment (JukkaL) [inline test-data/unit/check-incremental.test]: What's this?

File: test-data/unit/check-incremental.test [hunk 3]
```diff
@@ -4474,8 +4476,8 @@ def outer() -> None:
 [out]
 [out2]
 
-[case testRecursiveAliasImported]
-# flags: --no-new-semantic-analyzer
+[case testRecursiveAliasImported-skip]
```
Comment (JukkaL) [inline test-data/unit/check-incremental.test]: It looks like this test case can be removed, as the next case covers is roughly the same for the new semantic analyzer. Here the behavior in the new analyzer is different by design.

File: test-data/unit/check-incremental.test (hunk 2 above)
Comment (msullivan) [reply test-data/unit/check-incremental.test]: A typo

File: mypy/typeops.py (hunk 1 above)
Comment (msullivan) [reply mypy/typeops.py]: There is some fragility in mypy about import cycles, where certain modules need to be imported before others for things to work. Removing references to the old analyzer caused one of these places to surface when trying to run tests, so I fixed it. There are a couple more that I will try to fix in a separate PR.

Comment (JukkaL) [reply mypy/typeops.py]: One of these days I'm going to try to clean up the import cycles (though it might not help with this in particular).

File: mypy/build.py [hunk 4]
```diff
@@ -2024,31 +1989,22 @@ def semantic_analysis_pass1(self) -> None:
...
-            # import statements as unreachable.
-            first = SemanticAnalyzerPass1(self.manager.semantic_analyzer)
-            with self.wrap_context():
-                first.visit_file(self.tree, self.xpath, self.id, options)
+        # Do the first pass of semantic analysis: analyze the reachability
+        # of blocks and import statements. We must do this before
+        # processing imports, since this may mark some import statements as
+        # unreachable.
+        #
+        # TODO: Once we remove the old semantic analyzer, this no longer should
```
Comment (ilevkivskyi) [inline mypy/build.py]: This TODO item might be updated.

File: mypy/newsemanal/typeanal.py [hunk 5]
```diff
@@ -210,6 +210,7 @@ def visit_unbound_type_nonoptional(self, t: UnboundType, defining_literal: bool)
                 an_args = self.anal_array(t.args)
                 disallow_any = self.options.disallow_any_generics and not self.is_typeshed_stub
                 res = expand_type_alias(target, all_vars, an_args, self.fail, node.no_args, t,
+                                        unexpanded_type=t,
```
Comment (ilevkivskyi) [inline mypy/newsemanal/typeanal.py]: This change looks unrelated, or am I missing something?

File: test-data/unit/check-literal.test [hunk 6]
```diff
@@ -694,7 +694,7 @@ y: Foo[Foo]     # E: Literal[...] must have at least one parameter
 [out]
 
 [case testLiteralBadRawExpressionWithBadType]
-# flags: --new-semantic-analyzer
+
```
Comment (ilevkivskyi) [inline test-data/unit/check-literal.test]: Are these extra empty lines hard to avoid?

Comment (msullivan) [reply test-data/unit/check-literal.test]: They are easy to remove but they cause a lot of test failures due to line numbers changing and
1. Fixing them would have caused a lot more churn in the diff
2. I didn't want to fix them :P

File: mypy/build.py (hunk 4 above)
Comment (msullivan) [reply mypy/build.py]: Yeah

File: mypy/newsemanal/typeanal.py (hunk 5 above)
Comment (msullivan) [reply mypy/newsemanal/typeanal.py]: `expand_type_alias` is used by the type checker and switching the checker to using the new analyzer required some tweaks
